packages/explorer/src/app/(home)/EditableTableCell.tsx

```diff
@@ -0,0 +1,144 @@
+import { Loader } from "lucide-react";
+import { toast } from "sonner";
+import { privateKeyToAccount } from "viem/accounts";
+import { useChainId, useWriteContract } from "wagmi";
+import { ChangeEvent, useState } from "react";
+import { encodeField } from "@latticexyz/protocol-parser/internal";
+import { SchemaAbiType } from "@latticexyz/schema-type/internal";
+import { useQueryClient } from "@tanstack/react-query";
+import { waitForTransactionReceipt } from "@wagmi/core";
+import { Checkbox } from "../../components/ui/Checkbox";
+import { ACCOUNT_PRIVATE_KEYS } from "../../consts";
+import { useWorldAddress } from "../../hooks/useWorldAddress";
+import { camelCase } from "../../lib/utils";
+import { useStore } from "../../store";
+import { wagmiConfig } from "../_providers";
+import { TableConfig } from "../api/table/route";
+import { abi } from "./abi";
+import { getFieldIdx } from "./utils/getFieldIdx";
+
+type Props = {
+  name: string;
+  value: string;
+  keyTuple: string[];
+  config: TableConfig;
+};
+
+export function EditableTableCell({
+  name,
+  config,
+  keyTuple,
+  value: defaultValue,
+}: Props) {
+  const queryClient = useQueryClient();
+  const chainId = useChainId();
+  const { account } = useStore();
+  const { writeContractAsync } = useWriteContract();
+  const worldAddress = useWorldAddress();
+
+  const [loading, setLoading] = useState(false);
+  const [prevValue, setPrevValue] = useState<unknown>(defaultValue);
+  const [value, setValue] = useState<unknown>(defaultValue);
+
+  const tableId = config?.table_id;
+  const fieldType = config?.value_schema[camelCase(name)] as SchemaAbiType;
+
+  const handleSubmit = async (newValue: unknown) => {
```

Comment:
not blocking, but you might consider using a react-query mutation for this use case, which includes pending states, etc.

Reply:
refactored, although will still have to do some improvements here